baseservice/device_reader_service.py

```diff
@@ -68,20 +75,24 @@ def _server_loop(self, cancellation_token: threading.Event):
                 else:
                     timeout = 0  # if not wait_for_batch, try to read another message without waiting at all
 
-                message, device_headers = transaction_scope.read_message(timeout=timeout)
-                if message is None:
-                    break  # no more messages to read
+                result = transaction_scope.read_message(timeout=timeout)
+                if InputTransactionScope.is_non_empty_message(result):
+                    message, device_headers = result
```

Comment:
i don't like this (i know it's a duplicate comment on the review. but i don't like this twice...)